Packs/AWS-EC2/Integrations/AWS-EC2/AWS-EC2.py

```diff
@@ -2980,6 +2983,134 @@ def release_hosts_command(args, aws_client):
+    kwargs = {}
+    if (filters := args.get('Filters')) is not None:
+        kwargs.update({'Filters': parse_filter_field(filters)})
+    if (max_results := args.get('MaxResults')) is not None:
+        kwargs.update({'MaxResults': int(max_results)})
+    if (next_token := args.get('NextToken')) is not None:
+        kwargs.update({'NextToken': next_token})
+    if (ipam_ids := args.get('IpamResourceDiscoveryAssociationIds')) is not None:
+        kwargs.update({'IpamResourceDiscoveryAssociationIds': argToList(ipam_ids)})
+
+    response = client.describe_ipam_resource_discovery_associations(**kwargs)
+
+    if len(response['IpamResourceDiscoveryAssociations']) == 0:
+        return CommandResults(readable_output='No Ipam Resource Discovery Associations were found.')
+
+    human_readable = tableToMarkdown('Ipam Resource Discovery Associations', response['IpamResourceDiscoveryAssociations'])
+    command_results = CommandResults(
+        outputs_prefix="AWS.EC2.IpamResourceDiscoveryAssociations",
+        outputs_key_field="IpamResourceDiscoveryId",
+        outputs=response['IpamResourceDiscoveryAssociations'],
+        raw_response=response,
+        readable_output=human_readable,
+    )
+    return command_results
+
+
+def get_ipam_discovered_public_addresses_command(args: Dict[str, Any], aws_client) -> CommandResults:
+    """
+    aws-ec2-get-ipam-discovered-public-addresses: Gets the public IP addresses that have been discovered by IPAM.
+
+    Args:
+        client (AWSClient): AWS client to use.
+        args (dict): all command arguments, usually passed from ``demisto.args()``.
+
+    Returns:
+        CommandResults: A ``CommandResults`` object that is then passed to ``return_results``, that contains public IP addresses
+        that have been discovered by IPAM.
+    """
+    client = aws_client.aws_session(
+        service='ec2',
+        region=args.get('AddressRegion'),
+        role_arn=args.get('roleArn'),
+        role_session_name=args.get('roleSessionName'),
+        role_session_duration=args.get('roleSessionDuration')
+    )
+
+    if (args.get('IpamResourceDiscoveryId') is None) or (args.get('AddressRegion') is None):
+        return_error('IpamResourceDiscoveryId and AddressRegion need to be defined')
+
+    kwargs = {}
+    kwargs.update({'IpamResourceDiscoveryId': args.get('IpamResourceDiscoveryId'), 'AddressRegion': args.get('AddressRegion')})
+    if (filters := args.get('Filters')) is not None:
+        kwargs.update({'Filters': parse_filter_field(filters)})
+    if (max_results := args.get('MaxResults')) is not None:
+        kwargs.update({'MaxResults': int(max_results)})
+    if (next_token := args.get('NextToken')) is not None:
+        kwargs.update({'NextToken': next_token})
+
+    response = client.get_ipam_discovered_public_addresses(**kwargs)
+
+    if len(response['IpamDiscoveredPublicAddresses']) == 0:
+        return CommandResults(readable_output='No Ipam Discovered Public Addresseses were found.')
```

Comment:
```suggestion
        return CommandResults(readable_output='No Ipam Discovered Public Addresses were found.')
```
@amshamah419, can you fix this typo we just found, please?